tests/test_transfer_utils.py

```diff
@@ -0,0 +1,130 @@
+import sqlite3
+import pytest
+import segmentation_labeling_app.transfers.utils as utils
+import json
+import boto3
+from moto import mock_s3
+import pathlib
+import numpy as np
+import string
+
+
+@pytest.fixture(scope='module')
+def local_file(tmpdir_factory):
+    fn = tmpdir_factory.mktemp("files").join("test.txt")
+    with open(fn, "w") as fp:
+        fp.write("contents")
+    yield str(fn)
+
+
+@pytest.fixture(scope='module')
+def local_manifest(tmpdir_factory):
+    tdir = tmpdir_factory.mktemp("manifest_files")
+    manifest = {}
+    for i in range(3):
+        fname = tdir.join(f"test{i}.txt")
+        with open(fname, "w") as fp:
+            fp.write("contents")
+        manifest[f"key{i}"] = str(fname)
+    yield manifest
+
+
+@pytest.fixture(scope='module')
+def bucket():
+    with mock_s3():
+        bucket_name = 'mybucket'
+        conn = boto3.resource('s3')
+        conn.create_bucket(Bucket=bucket_name)
+        yield bucket_name
+
+
+@pytest.fixture(scope='function')
+def db_file(tmpdir_factory):
+    db = tmpdir_factory.mktemp("upload").join("mydatabase.db")
+    conn = sqlite3.connect(db)
+    conn.execute("CREATE TABLE manifest_table (manifest text)")
+    conn.close
+    yield db
+
+
+@pytest.mark.parametrize("key", ["abc/temp.csv", None])
+def test_upload_file(local_file, bucket, key):
```

Comment:
You can either use @mock_s3 or start the mock s3 service in the test to mock out all function calls to s3.

Reply:
It does work:
https://circleci.com/gh/AllenInstitute/segmentation-labeling-app/119

Reply:
Cool, didn't know that. Thanks!